### PR TITLE
Add board colors

### PR DESCRIPTION
**Pull request by christophermanahan**
### Description

Add a colorizer class of pure functions to add color codes to strings and utilize in formatter

board

### CODE REVIEW

File: lib/formatter.rb [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
     (1..board.size**2).to_a.map do |position|
       board.get(at: position)
     end
   end
 
-  def replace_nil_with_position(flattened)
+  def color_and_replace_nil(flattened)
     flattened.map.with_index do |symbol, position|
-      symbol.nil? ? position + 1 : symbol
+      case symbol
```
Comment (ssunday) [inline lib/formatter.rb]: Would you want a default color? Maybe the nil case, magenta, should be default?

Comment (christophermanahan) [reply lib/formatter.rb]: That is cleaner, thanks!

File: lib/formatter.rb [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+  attr_reader :colorizer, :symbols
+
   def flatten(board)
     (1..board.size**2).to_a.map do |position|
       board.get(at: position)
     end
   end
 
-  def replace_nil_with_position(flattened)
+  def color_and_replace_nil(flattened)
```
Comment (damonkelley) [inline lib/formatter.rb]: It feels like this method might have multiple responsibilities.

Comment (christophermanahan) [reply lib/formatter.rb]: That 'and' says it all.

**Comment by damonkelley** [comment]
@christophermanahan Can you include a screenshot of what the board looks like now?

File: lib/formatter.rb [hunk 3]
```diff
@@ -1,22 +1,43 @@
...
+    @symbols = symbols
+  end
+
   def format(board)
     flattened = flatten(board)
-    replaced = replace_nil_with_position(flattened)
-    rows = replaced.each_slice(board.size).to_a
+    symbols_or_positions = nil_to_position(flattened)
+    colored = color(symbols_or_positions)
+    rows = colored.each_slice(board.size).to_a
```
Comment (damonkelley) [inline lib/formatter.rb:11]: This method is getting kind of long.

**Review by damonkelley — APPROVED** [review]

**Pull request merged**